src/Symfony/Component/Routing/Tests/RouteTest.php

```diff
@@ -203,6 +203,21 @@ public function testSerialize()
         $this->assertNotSame($route, $unserialized);
     }
 
+    public function testInlineDefaultAndRequirement()
+    {
+        $this->assertEquals((new Route('/foo/{bar}'))->setDefault('bar', null), new Route('/foo/{bar?}'));
+        $this->assertEquals((new Route('/foo/{bar}'))->setDefault('bar', 'baz'), new Route('/foo/{bar?baz}'));
+        $this->assertEquals((new Route('/foo/{bar}'))->setDefault('bar', 'baz<buz>'), new Route('/foo/{bar?baz<buz>}'));
```

Comment:
How about we make it so that it does not matter if you specify a default or regex first?
So `'/foo/{bar?baz<.*>}' == '/foo/{bar<.*>?baz}'` 

This seems more user-friendly as you don't need to remember. Also this syntax is not feature-complete anyway do to missing escaping. So if you really need to specify a default like `baz<buz>` then just use the explicit config. But having such a default is much less likely than user mixing up the order.

Reply:
see https://github.com/symfony/symfony/issues/26481#issuecomment-372951223

Reply:
I think having two ways to express things just create more confusion.
Having only one order means everyone will write these in exactly the same way, thus less WTF when switching projects.

Reply:
Normally I agree consistency is better. But I also want to prevent bugs in the first place. So when we do not accept both ways, it would be cleaner to throw an exception if you do it wrong. I cannot believe somewhat actually uses a default like `baz<buz>` in a URL.

Reply:
> I cannot believe somewhat actually uses a default like baz<buz> in a URL

This is not a reason to technically forbid it. Symfony devs have been proved suprizing many time :)

Reply:
It's not forbidden. You just need to use the explicit syntax.

> Symfony devs have been proved surprising many time
Yes they will be surprised with unexpected behavior.